classes-recipe/image_types_qcom.bbclass

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) 2023-2024 Qualcomm Innovation Center, Inc. All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+
+inherit image_types
+
+IMAGE_TYPES += "qcomflash"
+
+QCOM_ESP_IMAGE ?= "esp-qcom-image"
+QCOM_ESP_FILE ?= "${@'efi.bin' if d.getVar('QCOM_ESP_IMAGE') else ''}"
+
+# There is currently no upstream-compatible way for the firmware to
+# identify and load the correct DTB from a combined-dtb that contains all
+# dtbs defined in KERNEL_DEVICETREE, so pick the first individual image
+# generated by linux-qcom-dtbbin, unless a different one is specified.
+QCOM_DTB_DEFAULT ?= "${@os.path.basename(d.getVar('KERNEL_DEVICETREE').split()[0][:-4]) if d.getVar('KERNEL_DEVICETREE') else ''}"
+QCOM_DTB_FILE ?= "dtb.bin"
+
+QCOM_ROOTFS_FILE ?= "system.img"
+IMAGE_QCOMFLASH_FS_TYPE ??= "ext4"
+
+QCOMFLASH_DIR = "${WORKDIR}/qcomflash"
+IMAGE_CMD:qcomflash = "create_qcomflash_pkg"
+do_image_qcomflash[dirs] = "${QCOMFLASH_DIR}"
+do_image_qcomflash[cleandirs] = "${QCOMFLASH_DIR}"
+do_image_qcomflash[depends] += "qcom-gen-partition-bins:do_deploy virtual/kernel:do_deploy \
+				${@'${QCOM_ESP_IMAGE}:do_image_complete' if d.getVar('QCOM_ESP_IMAGE') != '' else  ''}"
+IMAGE_TYPEDEP:qcomflash += "${IMAGE_QCOMFLASH_FS_TYPE}"
+
+create_qcomflash_pkg() {
+    # esp image
+    if [ -n "${QCOM_ESP_FILE}" ]; then
+        install -m 0644 ${DEPLOY_DIR_IMAGE}/${QCOM_ESP_IMAGE}-${MACHINE}.rootfs.vfat ${QCOM_ESP_FILE}
+    fi
+
+    # dtb image
+    if [ -n "${QCOM_DTB_DEFAULT}" ] && \
+                [ -f "${DEPLOY_DIR_IMAGE}/dtb-${QCOM_DTB_DEFAULT}-image.vfat" ]; then
+        # default image
+        install -m 0644 ${DEPLOY_DIR_IMAGE}/dtb-${QCOM_DTB_DEFAULT}-image.vfat ${QCOM_DTB_FILE}
+        # copy all images so they can be made available via the same tarball
+        for dtbimg in ${DEPLOY_DIR_IMAGE}/dtb-*-image.vfat; do
+            install -m 0644 ${dtbimg} .
+        done
+    fi
+
+    # rootfs image
+    install -m 0644 ${IMGDEPLOYDIR}/${IMAGE_LINK_NAME}.${IMAGE_QCOMFLASH_FS_TYPE} ${QCOM_ROOTFS_FILE}
+
+    # partition bins
+    for gptm in ${DEPLOY_DIR_IMAGE}/gpt_main*.bin; do
+        install -m 0644 ${gptm} .
+    done
+    for gpback in ${DEPLOY_DIR_IMAGE}/gpt_backup*.bin; do
+        install -m 0644 ${gpback} .
+    done
+    # skip BLANK_GPT and WIPE_PARTITIONS for rawprogram xml files
+    for rawpg in `find ${DEPLOY_DIR_IMAGE} -type f -name 'rawprogram*.xml' \
+                ! -name 'rawprogram*_BLANK_GPT.xml' ! -name 'rawprogram*_WIPE_PARTITIONS.xml'`; do
+        install -m 0644 ${rawpg} .
+    done
+    for patchfile in ${DEPLOY_DIR_IMAGE}/patch*.xml; do
```

Comment:
why are you mixing wildcard and find here, any reason not to do the same thing for consistency?

Reply:
Direct wildcard usage are files we know that are going to be available there, while for & find can be used to also cover the cases where they are not. Can move everything to use for & find for consistency, but it is not required.

Reply:
Migrated to a single for & find.